### PR TITLE
Feature/demands

### PR DESCRIPTION
**Pull request by arraisgabriel**
Implements demands and donors API

A demand is a necessity of a initiative, and a donor is a person who is donating something to fill this necessity.

The following paths were implemented:
```
/initiatives/{initiativeId}/demands [POST] [GET] 
/initiatives/{initiativeId}/demands/{demandId} [PUT] [DELETE] 
/initiatives/{initiativeId}/demands/{demandId}/donors [POST] [GET]
/initiatives/{initiativeId}/demands/{demandId}/donors/{donorId} [PUT] [DELETE] 
```

### CODE REVIEW

File: src/api/demand/controller.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    })
+  })
+    .then((initiative) => initiative ? initiative.view(true).demands : null)
+    .then(success(res, 201))
+    .catch(next)
+}
+
+export const index = ({ querymen: { query, select, cursor } }, res, next) =>
+  Demand.find(query, select, cursor)
+    .populate('creator donors donors.user')
```
Comment (diegohaz) [inline src/api/demand/controller.js]: Does `donors.user` work inside `.populate`? Shouldn't that be `.deepPopulate`?

File: src/api/demand/controller.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    .then((demands) => demands.map((demand) => demand.view(true)))
+    .then(success(res))
+    .catch(next)
+
+export const update = ({ user, bodymen: { body }, params }, res, next) =>
+  Demand.findById(params.demandId)
+    .populate('creator donors donors.user')
```
Comment (diegohaz) [inline src/api/demand/controller.js]: Same as above

File: src/api/initiative/controller.js [hunk 3]
```diff
@@ -14,22 +14,23 @@ export const create = ({ user, bodymen: { body } }, res, next) =>
...
+    .populate('user users tags demands')
+    .then((initiatives) => initiatives.map((initiative) => initiative.view(true)))
     .then(success(res))
     .catch(next)
 
 export const show = ({ params }, res, next) =>
   Initiative.findById(params.id)
-    .populate('user users tags')
+    .populate('user users tags demands')
+    .deepPopulate('demands.donors.user')
```
Comment (diegohaz) [inline src/api/initiative/controller.js]: I would use only `.deepPopulate('user users tags demands demands.donors.user')`

File: src/api/initiative/controller.js [hunk 4]
```diff
@@ -14,22 +14,23 @@ export const create = ({ user, bodymen: { body } }, res, next) =>
...
     .then(notFound(res))
     .then((initiative) => initiative ? initiative.view(true) : null)
     .then(success(res))
     .catch(next)
 
 export const update = ({ user, bodymen: { body }, params }, res, next) =>
   Initiative.findById(params.id)
-    .populate('user users tags')
+    .populate('user users tags demands demands.donors')
```
Comment (diegohaz) [inline src/api/initiative/controller.js]: Same as the two first comments. Does it work?

File: src/api/initiative/model.js [hunk 5]
```diff
@@ -4,6 +4,9 @@ import summary from 'node-summary'
 import { kebabCase } from 'lodash'
 import '../user'
 import '../tag'
+import '../demand'
+
+var deepPopulate = require('mongoose-deep-populate')(mongoose)
```
Comment (diegohaz) [inline src/api/initiative/model.js]: Please, use `const` instead of `var`.

File: src/api/demand/controller.js (hunk 1 above)
Comment (arraisgabriel) [inline src/api/demand/controller.js]: No it doesn't thanks for noting =)

**Pull request merged**